lib/digitizer_block_impl.cc

```diff
@@ -191,8 +191,8 @@ namespace gr {
      if (d_trigger_settings.direction == TRIGGER_DIRECTION_RISING
              || d_trigger_settings.direction == TRIGGER_DIRECTION_HIGH) {
 
-       float band = d_channel_settings[aichan].range / 100.0;
-       float lo = static_cast<float>(d_trigger_settings.threshold - band);
+       double band = d_channel_settings[aichan].range / 100.0;
```

Comment:
Maybe this is silly or I am missing a point: but wouldn't it be better to use `float` by default? 
It's easier to move from `float -> double` than the other way round.

We should include mp-units and a new C++ standard in gr-digitizers to get a better handle on type-safety.

Maby add additional `-Wall` and other sanitizers (clang-tidy, sonarlint) to the cmake files? This -- while initially annoying -- would produce warnings in places where the type-safety is violated and/or where there are ambiguities.

Reply:
> Maybe this is silly 

Indeed, I did too much, in this line it is actually not required to use double .. will keep it untouched.

> but wouldn't it be better to use float by default?

Actually I think both fload and double are not the right thing to do. Imo an enum would be best, since "range" anyhow only can have concrete values. (See #97) The type as well needs to be usable inside e.g. `digitizers_picoscope_3000a.xml`, so mp-units might not fit